### ISSUE TITLE
.NET 6 OData opens connection to SQL Server multiple times in get requests when using non-pooled connections

### ISSUE THREAD

**Issue by codingbunny23** [OData/AspNetCoreOData#845]
*Short summary (3-5 sentences) describing the issue.*
when we look at our outbound connections within the app service plan metrics on azure, we see that the amount of connections is anywhere from 3-12 times the amount of requests we had at the time. Meanwhile, our REST API's that look the exact same except they don't use OData returns a 1 to 1 relationship between outbound connections and requests.
### Assemblies affected
*Which assemblies and versions are known to be affected e.g. OData .Net lib 7.x*
 Microsoft.AspNetCore.OData v8.0.12 
 Microsoft.EntityFrameworkCore 6.0.13.
 net6.0
### Reproduce steps
*The simplest set of steps to reproduce the issue. If possible, reference a commit that demonstrates the issue.*
Setup an sql server on azure.
Setup a 2 endpoint net6.0 odata api with 1 odata endpoint with enablequery, and one non-odata endpoint.
Use our way of managing our connections due to non-data pooling, see additional detail.
Try to make a bunch of calls to odata and see the amount of outbound connections in the app service plan metrics compared to requests.
Try the same with non-odata endpoint.
### Expected result
*What would happen if there wasn't a bug.*
SQL connections being the only outbound connection of the request creates a 1 to 1 relationship between outbound connections per minute and requests per minute.
### Actual result
*What is actually happening.*
Outbound connections are 3-12 times that of requests.
### Additional detail
We have multiple REST API's using .NET 6 with Odata and EF.

In the projects we have `GET` endpoints that connect to a SQL Server on Azure. Those `GET` endpoints have `[EnableQuery]` to enable OData querying.

In the endpoints we create a database connection by creating a `DbContext `and then we asynchronously open the connection on the `.Database` parameter of the context object. We await the connection opening. Method used is `OpenConnectionAsync.`

This is to prevent EF/OData from opening a new connection each time a query is run, as we have thousands of databases we run through daily, and therefore pooling connections in our experience seems to have a net-negative effect on performance.

However, when we look at our outbound connections within the app service plan metrics on azure, we see that the amount of connections is anywhere from 3-12 times the amount of requests we had at the time. Meanwhile, our REST API's that look the exact same except they don't use OData returns a 1 to 1 relationship between outbound connections and requests.

In fact, when we took one of the OData API's and rewrote the endpoints with the most traffic to be non-OData we saw a massive decline in outbound connections.

Being unable to keep the amount of connections per request reasonable leads to the max amount of ip connections being exceeded per instance in azure. Leading to us having to increase the instance count and therefore the total cost per API. We would like to avoid that.

Anything I can do to tell OData to reuse the connection, as it doesn't seem to respect the OpenConnectionAsync?

**Comment by ElizabethOkerio** [comment]
@codingbunny23 Can you share with us a simple repro of the controller action method for both the non-odata and the odata APIs for us to investigate this further.

**Comment by codingbunny23** [comment]
@ElizabethOkerio Unfortunately I can't just post our source-code, as it would share too much closed-source company code.
However, your fair request for some code lead me to re-create the issue in a brand new small little test-API made simply for testing outbound connections and the difference between with OData and without.

 [Here is that repo:](https://github.com/codingbunny23/odataconnectiontest)

(you'll need a connection string to an SQL DB in startup.cs line 30).

(also our specific issue is outbound connections on azure app service plan for the app service the .net 6 app is running on, though the outbound connections issue may be general, it was uploading the code to an app service alone in an app service plan I used to test it)

Being the first time we have an API for this our customers aren't using. I could much easier be sure only the OData endpoint had been called within one time-period, and only non-OData within another. Thereby giving me a much easier time comparing data.

This lead to me discovering that the amount of outbound connections is in fact the same with the code I have uploaded. However, while after firing 100 request in 1 minute, the outbound connections fall back to normal level after that minute for the normal REST endpoint, when I used the OData endpoint the outbound connections never returned to a normal level within the next 15 minutes. In other words If I were to do another 100 requests next minute, the amount of outbound connections would just keep building. The problem isn't that it opens more connections than the regular REST version, but that it doesn't close the connections properly.

Which I found to be really odd considering I can't throw a `using` statement on the `context` without being given a 500 saying OData attempted to run `.Dispose` on an already disposed object. Therefore, I assume OData runs `.Dispose` at some point, and one would think the connections would close with it, afterall that is the technique we use in our normal REST endpoints. However, the numbers don't lie. Something causes it to not close properly.

Furthermore, I discovered that by removing `.OpenConnectionAsync `in the OData endpoint, it starts to close the connections properly again, and outbound connections returns to normal the minute after the requests. However, after removing the `.OpenConnectionAsync,` it now does use 3 times more connections than the regular REST version. Makes sense since we no longer explicitly open the connection to be reused throughout the SQL operations.

Still haven't found a way for it to both close it's connections as expected, and be comparable in connection usage to the non-OData version.
